integration-tests/checkoutcom_test.go

```diff
@@ -0,0 +1,187 @@
+package test
+
+import (
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/gateways/checkoutcom"
+	sleet_testing "github.com/BoltApp/sleet/testing"
+	"testing"
+	"time"
+)
+
+// TestCheckoutComAuthorizeFailed
+//
+// checkout.com has test cards here: https://www.checkout.com/docs/four/testing/response-code-testing
+// Using a rejected card number
+func TestCheckoutComAuthorizeFailed(t *testing.T) {
+	client := checkoutcom.NewClient(getEnv("CHECKOUTCOM_TEST_KEY"))
+	failedRequest := sleet_testing.BaseAuthorizationRequest()
+	failedRequest.CreditCard.Number = "4544249167673670"
+	response, err := client.Authorize(failedRequest)
+
+	if err != nil {
+		t.Errorf("Authorize request should not have an error even if authorization failed- %s", err.Error())
+	}
+
+	if response.Success {
+		t.Error("Auth response should indicate a failure")
+	}
+
+	if response.Response != "20051" {
+		t.Errorf("Response should be 20051, code for insufficient funds- %s", response.Response)
+	}
+}
+
+// TestCheckoutComAuth
+//
+// This should successfully create an authorization
+func TestCheckoutComAuth(t *testing.T) {
+	client := checkoutcom.NewClient(getEnv("CHECKOUTCOM_TEST_KEY"))
+	request := sleet_testing.BaseAuthorizationRequest()
+	auth, err := client.Authorize(request)
+	if err != nil {
+		t.Error("Authorize request should not have failed")
+	}
+
+	if !auth.Success {
+		t.Error("Resulting auth should have been successful")
+	}
+}
+
+// TestCheckoutComAuthFullCapture
+//
+// This should successfully create an authorization on checkout.com then Capture for full amount
+func TestCheckoutComAuthFullCapture(t *testing.T) {
+	client := checkoutcom.NewClient(getEnv("CHECKOUTCOM_TEST_KEY"))
+	authRequest := sleet_testing.BaseAuthorizationRequest()
+	auth, err := client.Authorize(authRequest)
+	if err != nil {
+		t.Error("Authorize request should not have failed")
+	}
+
+	if !auth.Success {
+		t.Error("Resulting auth should have been successful")
+	}
+
+	captureRequest := &sleet.CaptureRequest{
+		Amount:               &authRequest.Amount,
+		TransactionReference: auth.TransactionReference,
+		ClientTransactionReference: authRequest.ClientTransactionReference,
+	}
+	capture, err := client.Capture(captureRequest)
+	if err != nil {
+		t.Error("Capture request should not have failed")
+	}
+
+	if !capture.Success {
+		t.Error("Resulting capture should have been successful")
+	}
+}
+
+// TestCheckoutComAuthPartialCapture
+//
+// This should successfully create an authorization on checkout.com then Capture for a partial amount
+// Since we auth for 1.00USD, we will capture for $0.50
+func TestCheckoutComAuthPartialCapture(t *testing.T) {
```

Comment:
By including this test are we saying we support multicapture with checkout.com?

Reply:
Yeah I removed it. Won't be able to support until they (or we) updated the SDK to the capture request to add the right headers. Sounds like they'll be updating the SDK soon though.